spec/lib/stellar/sep10_spec.rb

```diff
@@ -473,6 +473,52 @@
         /The transaction is not signed by the server/
       )
     end
+
+    it "does not pass back duplicate signers or double-count weights" do
+      server_kp = Stellar::KeyPair.random
+      client_kp = Stellar::KeyPair.random
+
+      challenge_envelope = Stellar::TransactionEnvelope.from_xdr(
+        sep10.build_challenge_tx(
+          server: server_kp,
+          client: client_kp,
+          anchor_name: "SDF",
+          timeout: 600,
+        ), 
+        "base64"
+      )
+      challenge_envelope.signatures += [challenge_envelope.tx.sign_decorated(client_kp)]
+      challenge = challenge_envelope.to_xdr(:base64)
+
+      expect(
+        sep10.verify_challenge_tx_threshold(
+          challenge_xdr: challenge,
+          server: server_kp,
+          signers: Set[
+            {'key' => client_kp.address, 'weight' => 1},
+            {'key' => client_kp.address, 'weight' => 1}
+          ],
```

Comment:
This test isn't actually testing duplicate signers, it's testing a single signer because the `Set` type will automatically deduplicate these two entries. Take a look at [this repl](https://repl.it/repls/YawningIncompatibleScientificcomputing) as an example.

You should use an Array instead of a Set here so the duplicates are preserved outside of the SDK, and the SDK should be the one responsible for handling duplicates. Favor arrays in tests to ensure the SDK does the deduplication necessary.
```suggestion
          signers: [
            {'key' => client_kp.address, 'weight' => 1},
            {'key' => client_kp.address, 'weight' => 1}
          ],
```

Reply:
We can't pass arrays because the `Contract` will raise an exception if its not a `Set`, but I can use different weights so they're considered unique.

Reply:
When this happens, `verify_challenge_tx_threshold` will use the weight specified in the first hash seen. Duplicate signers' weights will be ignored.

Reply:
Ah, I missed that the Contract we've defined requires us to pass a Set. Got it!